docs/2.8.0/docs/daml/reference/templates.rst

```diff
@@ -92,7 +92,58 @@ Template-local Definitions
    - interface methods applied to ``this``
 
    in a template-local let definition will result in an error generated by the infinite loop.
- 
+
+.. warning::
+   Since Daml 2.8.0, template-local definitions are deprecated and their
+   presence will result in the following warning:
+
+   .. code-block:: text
+
+     Template-local binding syntax ("template-let") is deprecated,
+     it will be removed in a future version of Daml.
+     Instead, use plain top level definitions, taking parameters
+     for the contract fields or body ("this") if necessary.
+
+Migrating
+=========
+
+Users are strongly encouraged to rewrite their code to avoid this feature. Such
+a rewrite would involve replacing each template-local definition with a regular
+top level definition. If the old definition made use of contract fields or the
```

Comment:
```suggestion
Users are strongly encouraged to rewrite their code to avoid this feature. This involves replacing each template-local definition with a regular
top-level definition. If the old definition made use of contract fields or the
```